docs/guides/pagination/offset-based.md

```diff
@@ -0,0 +1,394 @@
+---
+title: Pagination - Offset based
+---
+
+# Implementing Offset Pagination
+
+Make sure to check our introduction to pagination [here](./overview.md)!
+
+Let us implement offset based pagination in GraphQL. By the end of this tutorial, we
+should be able to return a list of users which can be paginated by the client.
+
+```graphql+response
+query getUsers {
+  getUsers(offset: 0, limit: 2) {
+    users {
+      id
+      name
+      occupation
+      age
+    }
+    pageMeta {
+      total
+      pages
+    }
+  }
+}
+---
+{
+  "data": {
+    "getUsers": {
+      "users": [
+        {
+          "id": 1,
+          "name": "Norman Osborn",
+          "occupation": "Founder, Oscorp Industries",
+          "age": 42
+        },
+        {
+          "id": 2,
+          "name": "Peter Parker",
+          "occupation": "Freelance Photographer, The Daily Bugle",
+          "age": 16
+        }
+      ],
+      "pageMeta": {
+          "total": 4,
+          "page": 1,
+          "pages": 2
+      }
+    }
+  }
+}
+```
+
+Let us model our schema like this:
+
+```py
+# example.py
+
+from typing import List
+
+import strawberry
+
+
+@strawberry.type
+class User:
+    name: str = strawberry.field(
+        description="""
+        The name of the user.
+        """
+    )
+
+    occupation: str = strawberry.field(
+        description="""
+        The occupation of the user.
+        """
+    )
+
+    age: int = strawberry.field(
+        description="""
+        The age of the user.
+        """
+    )
+
+
+@strawberry.type
+class PageMeta:
+    total: int = strawberry.field(
+        description="""
+        The total number of items in the dataset.
+        """
```

Comment:
why are are all of these triple quote strings with multiple line breaks?

Reply:
It's just to keep uniformity, rather a personal preference. I think the description is more readable this way. I can remove that if needed.

Reply:
Yes, let's just use standard strings, please

Reply:
done! but this messes up code highlight 😢

Reply:
I finally fixed code highlight!